src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,202 @@
+                } else if (!input.contains("/")) {
+                    throw new DukeException("OOPS!!! Your event command's format is invalid. " + validFormat);
+                } else if (input.substring(6).split("/").length < 3) {
+                    throw new DukeException("OOPS!!! Your event command's format is invalid. " + validFormat);
+                } else {
+                    String[] parts = input.substring(6).split("/");
+                    LocalDateTime dateTime1;
+                    LocalDateTime dateTime2;
+                    try {
+                        String dateTimeString1 = parts[1].substring(5).trim();
+                        String dateTimeString2 = parts[2].substring(3).trim();
+                        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+                        dateTime1 = LocalDateTime.parse(dateTimeString1, formatter);
+                        dateTime2 = LocalDateTime.parse(dateTimeString2, formatter);
+                    } catch (DateTimeParseException e) {
+                        throw new DukeException("OOPS!!! The date format is invalid. " + validFormat);
+                    }
+                    Events eventTask = new Events(parts[0], dateTime1, dateTime2);
+                    AddTask adder = new AddTask(storage, eventTask);
+                    adder.add();
+                    return adder.toString();
+                }
+            } else if (input.startsWith("delete")) {
+                String validFormat = "Please input in format: delete *index*";
+                if (input.trim().equals("delete")) {
+                    throw new DukeException("OOPS!!! Invalid delete command " + validFormat);
+                } else {
+                    int index = Integer.parseInt(input.substring(7));
+                    if (index < 1 || index > inventory.size()) {
+                        throw new DukeException("OOPS!!! The task number you are trying to delete does not exist. ");
+                    }
+                    DeleteTask deleter = new DeleteTask(storage, index);
+                    deleter.delete();
+                    return deleter.toString();
+                }
+            } else if (input.startsWith("find")) {
+                String validFormat = "Please input in format: find *Task*";
+                if (input.trim().equals("find")) {
+                    throw new DukeException("OOPS!! Invalid find command " + validFormat);
+                } else {
+                    String task = input.substring(5);
+                    FindTask finder = new FindTask(storage, task);
+                    finder.createMatches();
+                    return finder.toString();
+                }
+            } else {
+                return "OOPS!!! I'm sorry, but that's an invalid command :-(";
+            }
+        } catch (DukeException e) {
+            return e.getMessage();
+        }

```

Comment:
Maybe consider letting the output class handle the printing of exceptions? Since supposedly that class is responsible for all outputs to the users.